pages/api/book/[isbn].ts

```diff
@@ -0,0 +1,32 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { HttpMethod } from "@lib/http-method";
+import { StatusCodes } from "http-status-codes";
+import { getBookWithPosts } from "../../../lib/services/book";
+import { isValidISBN } from "../../../lib/helpers/backend/valid-isbn";
+
+export default function handler(req: NextApiRequest, res: NextApiResponse) {
+  switch (req.method as HttpMethod) {
+    case HttpMethod.GET:
+      return getBookWithPostsHandler(req, res);
+    default:
+      return res.status(StatusCodes.METHOD_NOT_ALLOWED).end();
+  }
+}
+
+async function getBookWithPostsHandler(req: NextApiRequest, res: NextApiResponse) {
+  const { isbn } = req.query;
+  const bookISBN = isbn as string;
```

Comment:
This isn't type safe, I'd recommend doing something like this instead:
```
  const { isbn } = req.query;

  if (Array.isArray(isbn)) {
    return res.status(StatusCodes.BAD_REQUEST).end();
  }
```

Reply:
Will be included in the PR. Thanks!